File: demo/src/main/java/kaufland/com/demo/MainActivity.java
package kaufland.com.demo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ListView;

import com.couchbase.lite.DataSource;
import com.couchbase.lite.Database;
import com.couchbase.lite.Expression;
import com.couchbase.lite.Query;
import com.couchbase.lite.QueryBuilder;
import com.couchbase.lite.Result;
import com.couchbase.lite.SelectResult;

import kaufland.com.demo.entity.ProductEntity;

public class MainActivity extends AppCompatActivity {

    private ProductAdapter mAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        mAdapter = new ProductAdapter(this);

        mAdapter.addAll(ProductEntity.findByType());
        mAdapter.notifyDataSetChanged();

        ListView listView = findViewById(R.id.list);
        listView.setAdapter(mAdapter);
        listView.setOnItemClickListener((parent, view, position, id) -> {
            ProductEntity mItem = mAdapter.getItem(position);
            startActivity(CommentActivity.buildIntent(MainActivity.this, mItem.getId()));
        });
    }

}

File: demo/src/main/java/kaufland/com/demo/MainActivity.kt
package kaufland.com.demo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.ListView
import kaufland.com.demo.CommentActivity.Companion.buildIntent
import kaufland.com.demo.entity.ProductEntity.Companion.findByType

class MainActivity : AppCompatActivity() {
    private var mAdapter: ProductAdapter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        mAdapter = ProductAdapter(this)
        mAdapter!!.addAll(findByType())
        mAdapter!!.notifyDataSetChanged()
        val listView = findViewById<ListView>(R.id.list)
        listView.adapter = mAdapter
        listView.onItemClickListener = OnItemClickListener { parent: AdapterView<*>?, view: View?, position: Int, id: Long ->
            val mItem = mAdapter!!.getItem(position)
            startActivity(buildIntent(this@MainActivity, mItem!!.getId()))
        }
    }
}
